internal/interpreter/json.rs

```diff
@@ -86,14 +86,85 @@ pub fn value_from_json(t: &langtype::Type, v: &serde_json::Value) -> Result<Valu
+                            } else {
+                                let color = string_to_color(sub_parts[0]);
+                                let position = {
+                                    if let Some(percent_value) = sub_parts[1].strip_suffix("%") {
+                                        percent_value.parse::<f32>().map_err(|_| format!("Could not parse position '{}' as number", sub_parts[1]))
+                                    } else {
+                                        Err(format!("The position '{}' doea not end in '%'", sub_parts[1]))
+                                    }
+                                };
+
+                                match (color, position) {
+                                    (Some(c), Ok(p)) => Ok(i_slint_core::graphics::GradientStop { color: c, position: p / 100.0}),
+                                    (_, Err(e)) => Err(e),
+                                    (None, _) => Err(format!("'{}' is not a color", sub_parts[0])),
+                                }
+                            }
+                        }).collect()
+                    }
+
+                    let Some(input) = input.strip_suffix(')') else {
+                        return Err(format!("No closing ')' in '{input}'"));
+                    };
+
+                    if let Some(linear) = input.strip_prefix("@linear-gradient(") {
+                        let mut split = linear.split(',').map(|p| p.trim());
+
+                        let angle = {
+                            let Some(angle_part) = split.next() else {
+                                return Err(
+                                    "A linear gradient must start with an angle in 'deg'".into()
+                                );
+                            };
+
+                            angle_part
+                                .strip_suffix("deg")
+                                .ok_or_else(|| {
+                                    "A linear brush needs to start with an angle in 'deg'"
+                                        .to_string()
+                                })
+                                .and_then(|no| {
+                                    no.parse::<f32>()
+                                        .map_err(|_| "Failed to parse angle value".into())
+                                })
+                        }?;
+
+                        Ok(i_slint_core::graphics::LinearGradientBrush::new(
+                            angle,
+                            parse_stops(split)?.drain(..),
+                        )
+                        .into())
+                    } else if let Some(radial) = input.strip_prefix("@radial-gradient(circle") {
+                        let split = radial.split(',').map(|p| p.trim());
+
+                        Ok(i_slint_core::graphics::RadialGradientBrush::new_circle(
+                            parse_stops(split)?.drain(..),
```

Comment:
why `drain(..)` and not into_iter or something like that?

Reply:
Because it needs a `Iterator<Item = GradientStop>` and that is a good way to get that out of a Vec without creating yet another copy.